web/api/v2/create-action/index.ts

```diff
@@ -0,0 +1,187 @@
+import { errorResponse } from "@/api/helpers/errors";
+import {
+  getAPIKeyGraphqlClient,
+  getAPIServiceGraphqlClient,
+} from "@/api/helpers/graphql";
+import { verifyHashedSecret } from "@/api/helpers/utils";
+import { validateRequestSchema } from "@/api/helpers/validate-request-schema";
+import { generateExternalNullifier } from "@/lib/hashing";
+import { ApolloError } from "@apollo/client";
+import { NextRequest, NextResponse } from "next/server";
+import * as yup from "yup";
+import {
+  CreateDynamicActionMutation,
+  getSdk as createDynamicActionSdk,
+} from "./graphql/create-dynamic-action.generated";
+import { getSdk as fetchApiKeySdk } from "./graphql/fetch-api-key.generated";
+
+const createActionBodySchema = yup.object({
+  action: yup.string().strict().required(),
+});
+
+const createActionParamsSchema = yup.object({
+  app_id: yup.string().strict().required(),
+});
+
+export const POST = async (
+  req: NextRequest,
+  { params: rawParams }: { params: { app_id: string } },
+) => {
+  const rawBody = await req.json();
+  const api_key = req.headers.get("authorization")?.split(" ")[1];
+
+  if (!api_key) {
+    return errorResponse({
+      statusCode: 401,
+      code: "unauthorized",
+      detail: "API key is required.",
+      attribute: "api_key",
+      req,
+    });
+  }
+
+  const {
+    isValid: isParamsValid,
+    parsedParams: params,
+    handleError: handleParamsValidationError,
+  } = await validateRequestSchema({
+    schema: createActionParamsSchema,
+    value: rawParams,
+  });
+
+  if (!isParamsValid) {
+    return handleParamsValidationError(req);
+  }
+
+  const {
+    isValid: isBodyValid,
+    parsedParams: body,
+    handleError: handleBodyValidationError,
+  } = await validateRequestSchema({
+    schema: createActionBodySchema,
+    value: rawBody,
+  });
+
+  if (!isBodyValid) {
+    return handleBodyValidationError(req);
+  }
+
+  const { action, app_id } = {
+    ...body,
+    ...params,
+  };
+
+  const keyValue = api_key.replace(/^api_/, "");
+  const base64ApiKey = Buffer.from(keyValue, "base64").toString("utf8");
+  const [id, secret] = base64ApiKey.split(":");
+  const serviceClient = await getAPIServiceGraphqlClient();
```

Comment:
@andy-t-wang Had to use service client here to fetch API key data, as we agreed to get team_id from the key id. 

Without team_id we can't create api_key client because we need team_id in hasura jwt payload.

If we don't want to use service client at all in this endpoint we need to accept team_id via body